test/extended/cpu_partitioning/crio.go

```diff
@@ -321,22 +324,29 @@ func collectContainerInfo(ctx context.Context, oc *exutil.CLI, node corev1.Node)
 
 // Due to cluster activity, sometimes execing against a pod can be flakey.
 // We retry if the below errors are received during node queries.
-//
-// JSON Syntax error:
-//
-//	This typically means that something interrupted the stream of json coming back, so we need to do the query again.
-//
-// Dialing timeout:
-//
-//	Some sort of disruption is happening causing the exec to time out, retry.
 func shouldRetryExec(err error) bool {
 	var parseErr *json.SyntaxError
+	substringErrors := []string{
+		"error dialing backend: dial tcp", // Some sort of disruption is happening causing the exec to time out, retry.
+		"container not found",             // On occasion there might be situations where the machine-config-daemon isn't running yet.
+		"err execing command",             // Error when running the command in the container
+	}
 	switch {
+	// This typically means that something interrupted the stream of json coming back, so we need to do the query again.
 	case errors.As(err, &parseErr):
 		return true
-	case strings.Contains(err.Error(), "error dialing backend: dial tcp"):
+	case substringSliceContains(substringErrors, err.Error()):
```

Comment:
can we use `errors.As` also here? compare errors with known values? matching the error string should really be the last resort.

Reply:
I did try and dig through the return types, unfortunately I didn't find a great way to get a guarantee on the error type being thrown there, so the string seemed to be the next sure thing.